			   READLINE PATCH REPORT
			   =====================

Readline-Release: 8.1
Patch-ID: readline81-002

Bug-Reported-by:	Volodymyr Prodan <vovcat@gmail.com>
Bug-Reference-ID:
Bug-Reference-URL:	https://savannah.gnu.org/patch/?10076

Bug-Description:

There are some characters (e.g., cyrillic) that can't be displayed using
certain single-byte encodings (e.g., cp1251) because the negative signed
int is interpreted as EOF and not displayed.

Patch (apply with `patch -p0'):

*** ../readline-8.1-patched/display.c	2021-03-16 18:12:20.000000000 -0400
--- display.c	2021-06-07 16:53:08.000000000 -0400
***************
*** 1599,1603 ****
  
    for (cur_face = FACE_NORMAL, i = 0; i < n; i++)
!     putc_face (str[i], face[i], &cur_face);
    putc_face (EOF, FACE_NORMAL, &cur_face);
  }
--- 1599,1603 ----
  
    for (cur_face = FACE_NORMAL, i = 0; i < n; i++)
!     putc_face ((unsigned char) str[i], face[i], &cur_face);
    putc_face (EOF, FACE_NORMAL, &cur_face);
  }

*** ../readline-8.1/patchlevel	2013-11-15 08:11:11.000000000 -0500
--- patchlevel	2014-03-21 08:28:40.000000000 -0400
***************
*** 1,3 ****
  # Do not edit -- exists only for use by patch
  
! 1
--- 1,3 ----
  # Do not edit -- exists only for use by patch
  
! 2
